public/app/features/explore/ExploreToolbar.tsx

```diff
@@ -76,7 +87,21 @@ export function ExploreToolbar({ exploreId, topOfViewRef, onChangeTime }: Props)
   };
 
   const onChangeDatasource = async (dsSettings: DataSourceInstanceSettings) => {
-    dispatch(changeDatasource(exploreId, dsSettings.uid, { importQueries: true }));
+    // left pane datasource changes don't affect created correlation. Right pane datasource changes might be destructive
```

Comment:
this is something i found a bit confusing when playing with the feature. once i change the left datasource what's the source of the correlation? the new or the old one?

Reply:
You're right, this doesn't make sense. I changed it so it should do the following

If you change the datasource on either side and there is a dirty state, prompt for save/discard/cancel. Discarding will change the datasource after.

If there is no dirty state, remove the helper data. It will still be in split mode.

This makes sure the helper will always be consistent with the datasource on the left side. Changing the left side in other ways - deleting/editing/hiding a query - will not affect the helper because the datasource itself isn't changing and that is the important relationship to have with the helper.